l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+        config.contracts.diamondInitPriorityTxMaxPubdata = toml.readUint(
+            "$.contracts.diamond_init_priority_tx_max_pubdata"
+        );
+        config.contracts.diamondInitMinimalL2GasPrice = toml.readUint("$.contracts.diamond_init_minimal_l2_gas_price");
+
+        config.tokens.tokenWethAddress = toml.readAddress("$.tokens.token_weth_address");
+    }
+
+    function instantiateCreate2Factory() internal {
+        address contractAddress;
+
+        bool isDeterministicDeployed = DETERMINISTIC_CREATE2_ADDRESS.code.length > 0;
+        bool isConfigured = config.contracts.create2FactoryAddr != address(0);
+
+        if (isConfigured) {
+            if (config.contracts.create2FactoryAddr.code.length == 0) {
+                revert("Create2Factory configured address is empty");
+            }
+            contractAddress = config.contracts.create2FactoryAddr;
+            console.log("Using configured Create2Factory address:", contractAddress);
+        } else if (isDeterministicDeployed) {
+            contractAddress = DETERMINISTIC_CREATE2_ADDRESS;
+            console.log("Using deterministic Create2Factory address:", contractAddress);
+        } else {
+            contractAddress = Utils.deployCreate2Factory();
+            console.log("Create2Factory deployed at:", contractAddress);
+        }
+
+        addresses.create2Factory = contractAddress;
+    }
+
+    function deployIfNeededMulticall3() internal {
+        // Multicall3 is already deployed on public networks
+        if (MULTICALL3_ADDRESS.code.length == 0) {
+            address contractAddress = deployViaCreate2(type(Multicall3).creationCode);
+            console.log("Multicall3 deployed at:", contractAddress);
+        }
+    }
+
+    function deployVerifier() internal {
+        address contractAddress = deployViaCreate2(type(Verifier).creationCode);
+        console.log("Verifier deployed at:", contractAddress);
+        addresses.stateTransition.verifier = contractAddress;
+    }
+
+    function deployDefaultUpgrade() internal {
+        address contractAddress = deployViaCreate2(type(DefaultUpgrade).creationCode);
+        console.log("DefaultUpgrade deployed at:", contractAddress);
+        addresses.stateTransition.defaultUpgrade = contractAddress;
+    }
+
+    function deployGenesisUpgrade() internal {
+        address contractAddress = deployViaCreate2(type(GenesisUpgrade).creationCode);
+        console.log("GenesisUpgrade deployed at:", contractAddress);
+        addresses.stateTransition.genesisUpgrade = contractAddress;
+    }
+
+    function deployValidatorTimelock() internal {
+        uint32 executionDelay = uint32(config.contracts.validatorTimelockExecutionDelay);
+        bytes memory bytecode = abi.encodePacked(
+            type(ValidatorTimelock).creationCode,
+            abi.encode(config.deployerAddress, executionDelay, config.eraChainId)
```

Comment:
the first parameter is `owner address`

Reply:
Based on the fact that we have to make some manipulations with it, i'd prefer to set owner at the end of the script, for not switching private keys in the middle of the process